addons/web_dashboard/static/src/js/dashboard.js

```diff
@@ -0,0 +1,264 @@
+            var deferred_promises = []
+            _.each(dashboards, function(dashboard) {
+                var def = self[dashboard](data);
+                if (def) {
+                    deferred_promises.push(def);
+                }
+                framework.unblockUI();
+            });
+            $.when.apply($, deferred_promises)
+                .then(function() {
+                    loading_done.resolve();
+                });
+        });
+        return loading_done;
+    },
+    apps: function(data){
+        this.$('.o_web_dashboard_apps').append(QWeb.render("DashboardApps", data['apps']));
+    },
+    share: function(data){
+        return new DashboardShare(this, {}).replace(this.$('.o_web_dashboard_share'));
+    },
+    invitations: function(data){
+        return new DashboardInvitations(this, data['users_info']).replace(this.$('.o_web_dashboard_invitations'));
+    },
+    planner: function(data){
+        return  new DashboardPlanner(this, data['planner']).replace(this.$('.o_web_dashboard_planner'));
+    },
+    gift: function(data){
+        return new DashboardGift(this, data['gift']).replace(this.$('.o_web_dashboard_gift'));
+    },
+    new_apps: function(){
+        this.do_action('base.open_module_tree', {
+            'additional_context': {'search_default_app': 1, 'search_default_not_installed': 1}
+        });
+    }
+});
+
+var DashboardInvitations = Widget.extend({
+    template: 'DashboardInvitations',
+    events: {
+        'click .o_send_invitations': 'send_invitations',
+        'click .optional_message_toggler': 'optional_message_toggler',
+        'click .user': 'on_user_clicked',
+    },
+    init: function(parent, data){
+        this.data = data;
+        this.parent = parent;
+        return this._super.apply(this, arguments);
+    },
+    send_invitations: function(e){
+        var self = this;
+        var $target = $(e.currentTarget);
+        var user_emails =  _.filter(this.$('#user_emails').val().split(/[\n, ]/), function(email){
+            return email != "";
+        });
+        var re = /^([\w-]+(?:\.[\w-]+)*)@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$/i;
+        var is_valid_emails = _.every(user_emails, function(email) {
+            return re.test(email);
+        });
+        var optional_message = this.$('#optional_message').val();
+        if(is_valid_emails){
+            $target.prop('disabled', true);
+            $target.find('i.fa-cog').removeClass('hidden');
+            new Model("res.users")
+                .call("web_dashboard_create_users", [user_emails, optional_message])
+                .then(function (result) {
+                    self.reload();
+                });
+
+        }else{
+            this.do_warn(_t("Please provide valid email addresses"), "");
+        }
+    },
+    on_user_clicked: function (event) {
+        var self = this;
+        var user_id = $(event.target).data('user-id');
```

Comment:
Element on which event is attached don't having any child so it doesn't matter `target or currentTarget` because both going to return same result